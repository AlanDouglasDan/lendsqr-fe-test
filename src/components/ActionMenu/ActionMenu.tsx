import { Link } from "react-router-dom";

import {
  EyeIcon,
  DeleteUserIcon,
  UserTimesIcon,
  ActivateUserIcon,
} from "components/icons";
import "./ActionMenu.styles.scss";

type Props = {
  setStatusesArray: (data: any) => void;
  statusesArray: Array<JSX.Element>;
  status: JSX.Element;
  userId: string;
  index: string;
  blacklisted: JSX.Element;
  active: JSX.Element;
  inactive: JSX.Element;
};

const ActionMenu: React.FC<Props> = ({
  setStatusesArray,
  status,
  statusesArray,
  index,
  userId,
  blacklisted,
  active,
  inactive,
}) => {
  return (
    <div className="dropdown">
      <i
        className="fa fa-ellipsis-v ellipsisIcon cursor-pointer dropdown-toggle"
        data-bs-toggle="dropdown"
      />
      <ul className="dropdown-menu">
        <Link to={`/dashboard/users/${userId}`} className="no-underline text-dark">
          <li className="dropdown-item align-items-center d-flex py-2">
            <EyeIcon />
            <span className="ms-2"> View Details</span>
          </li>
        </Link>
        {status !== blacklisted && (
          <li
            className="dropdown-item align-items-center d-flex py-2"
            onClick={() => {
              setStatusesArray(
                statusesArray.map((status: any, i: any) =>
                  i === index ? blacklisted : status
                )
              );
            }}
          >
            <DeleteUserIcon /> <span className="ms-2">Blacklist User</span>
          </li>
        )}
        {status === active ? (
          <li
            className="dropdown-item align-items-center d-flex py-2"
            onClick={() => {
              setStatusesArray(
                statusesArray.map((status: any, i: any) =>
                  i === index ? inactive : status
                )
              );
            }}
          >
            <UserTimesIcon /> <span className="ms-2">Deactivate User</span>
          </li>
        ) : (
          <li
            className="dropdown-item align-items-center d-flex py-2"
            onClick={() => {
              setStatusesArray(
                statusesArray.map((status: any, i: any) =>
                  i === index ? active : status
                )
              );
            }}
          >
            <ActivateUserIcon /> <span className="ms-2">Activate User</span>
          </li>
        )}
      </ul>
    </div>
  );
};

export default ActionMenu;
