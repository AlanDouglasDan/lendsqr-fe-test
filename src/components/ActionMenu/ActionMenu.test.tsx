import { BrowserRouter } from "react-router-dom";
import { render } from "@testing-library/react";

import ActionMenu from "./ActionMenu";

const pending = <span className="light-yellow-btn">Pending</span>;
const active = <span className="light-green-btn">Active</span>;
const inactive = <span className="light-grey-btn">Inactive</span>;
const blacklisted = <span className="light-red-btn">Blacklisted</span>;

describe("Action Menu Component", () => {
  it("Using snapshot testing", () => {
    const { asFragment } = render(
      <BrowserRouter>
        <ActionMenu
          userId="1"
          status={pending}
          setStatusesArray={() => {}}
          index="0"
          statusesArray={[pending, active, inactive, blacklisted]}
          blacklisted={blacklisted}
          active={active}
          inactive={inactive}
        />
      </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
