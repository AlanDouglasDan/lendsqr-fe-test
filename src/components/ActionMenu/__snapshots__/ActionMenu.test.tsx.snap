// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Action Menu Component Using snapshot testing 1`] = `
<DocumentFragment>
  <div
    class="dropdown"
  >
    <i
      class="fa fa-ellipsis-v ellipsisIcon cursor-pointer dropdown-toggle"
      data-bs-toggle="dropdown"
    />
    <ul
      class="dropdown-menu"
    >
      <a
        class="no-underline text-dark"
        href="/dashboard/users/1"
      >
        <li
          class="dropdown-item align-items-center d-flex py-2"
        >
          <svg
            fill="none"
            height="12"
            viewBox="0 0 16 12"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M15.4533 5.44011L15.4519 5.43845C15.0398 4.92184 14.0948 3.82505 12.7977 2.85586C11.4993 1.88564 9.83832 1.03611 7.99968 1.03611C6.16104 1.03611 4.50011 1.88561 3.20166 2.85582C1.9029 3.82627 0.957157 4.92466 0.545819 5.44047C0.274826 5.76662 0.277249 6.2343 0.544833 6.57367L0.544827 6.57368L0.545641 6.57468C0.956296 7.08187 1.90229 8.17692 3.20172 9.14589C4.50012 10.1141 6.16105 10.9636 7.99968 10.9636C9.83832 10.9636 11.4993 10.1141 12.7977 9.1438C14.0966 8.17315 15.0424 7.07445 15.4537 6.55838C15.7074 6.2495 15.7071 5.74924 15.4533 5.44011ZM7.99968 9.75611C6.48691 9.75611 5.06807 9.02252 3.92942 8.17201C2.84501 7.36201 2.02502 6.4537 1.63351 5.9981C2.01625 5.53083 2.83628 4.6224 3.92306 3.81583C5.06351 2.96943 6.48657 2.24347 7.99968 2.24347C9.51274 2.24347 10.9317 2.96936 12.0701 3.81576C13.1557 4.62284 13.9761 5.53202 14.3662 5.99979C13.9762 6.46752 13.1557 7.3767 12.0701 8.18379C10.9317 9.0302 9.51274 9.75611 7.99968 9.75611Z"
              fill="#545F7D"
              stroke="#545F7D"
              stroke-width="0.2"
            />
            <path
              d="M8.00014 2.90818C6.29675 2.90818 4.9083 4.2967 4.9083 6.00002C4.9083 7.70334 6.29682 9.09186 8.00014 9.09186C9.70346 9.09186 11.092 7.70334 11.092 6.00002C11.092 4.29669 9.70346 2.90818 8.00014 2.90818ZM8.00014 7.88386C6.96726 7.88386 6.11646 7.0324 6.11646 6.00018C6.11646 4.96728 6.96732 4.1165 8.00014 4.1165C9.03296 4.1165 9.88382 4.96736 9.88382 6.00018C9.88382 7.033 9.03296 7.88386 8.00014 7.88386Z"
              fill="#545F7D"
              stroke="#545F7D"
              stroke-width="0.2"
            />
          </svg>
          <span
            class="ms-2"
          >
             View Details
          </span>
        </li>
      </a>
      <li
        class="dropdown-item align-items-center d-flex py-2"
      >
        <svg
          fill="none"
          height="14"
          viewBox="0 0 12 14"
          width="12"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M5.08321 5.78713L5.08358 5.7875H5.125C5.84771 5.7875 6.54087 5.50023 7.05183 4.9893L6.98112 4.91859L7.05183 4.9893C7.56276 4.47837 7.85003 3.78519 7.85003 3.06247C7.85003 2.33974 7.56276 1.64659 7.05183 1.13564L6.98516 1.20231L7.05183 1.13564C6.5409 0.624706 5.84772 0.337439 5.125 0.337439C4.40227 0.337439 3.70913 0.624707 3.19817 1.13563L3.19817 1.13564C2.68724 1.64657 2.39997 2.33974 2.39997 3.06247V3.06255C2.40054 3.78527 2.68838 4.47782 3.19867 4.9887L3.19875 4.98878C3.69979 5.48925 4.37558 5.77575 5.08321 5.78713ZM3.47503 3.06258C3.47606 2.15406 4.20966 1.41783 5.11717 1.41249H5.125C5.79221 1.41249 6.39395 1.81469 6.64919 2.43084L6.64921 2.43087C6.90495 3.04756 6.76367 3.7576 6.29189 4.22935L6.3626 4.30007L6.29189 4.22936C5.8201 4.70114 5.11009 4.84242 4.49341 4.58668L4.49337 4.58666C3.87725 4.33145 3.47507 3.72976 3.47503 3.06258ZM0.36976 13.5052C0.470651 13.6061 0.607691 13.6625 0.749994 13.6625H9.49999C9.6423 13.6625 9.77935 13.6061 9.88023 13.5052C9.98112 13.4043 10.0375 13.2673 10.0375 13.125L10.0375 11.375L10.0375 11.3749C10.0358 10.0725 9.51779 8.82375 8.5973 7.90272L8.59726 7.90268C7.67619 6.98218 6.4274 6.46421 5.12512 6.4625L5.12504 6.52398L5.125 6.46249H5.12499L5.12495 6.52296L5.12487 6.4625C3.82254 6.46417 2.57375 6.98219 1.65273 7.90268L1.65269 7.90273C0.732181 8.82379 0.214215 10.0726 0.2125 11.3749V11.375V13.125C0.2125 13.2673 0.268865 13.4043 0.36976 13.5052ZM8.96249 12.5875H1.2875V11.375C1.2875 10.0041 2.01876 8.73726 3.20617 8.0519L3.2062 8.05189C4.39368 7.36594 5.85619 7.36594 7.04366 8.05189L7.04369 8.0519C8.23117 8.73731 8.96236 10.0043 8.96236 11.375L8.96249 12.5875ZM11.2446 6.23463L10.6974 5.68751L11.2452 5.1397L11.2452 5.13971L11.2464 5.13854C11.4507 4.92746 11.4472 4.59222 11.2404 4.3848L11.2402 4.38457C11.0328 4.1778 10.6975 4.17427 10.4865 4.37863L10.4853 4.37976L9.93749 4.92757L9.38969 4.37976L9.3897 4.37975L9.38853 4.37863C9.17744 4.17426 8.84221 4.17782 8.63479 4.38457L8.63456 4.3848C8.42779 4.59222 8.42426 4.92746 8.62861 5.13854L8.6286 5.13855L8.62975 5.1397L9.17756 5.68751L8.62975 6.23532L8.62974 6.23531L8.62862 6.23647C8.42425 6.44756 8.4278 6.78279 8.63456 6.99022L8.63478 6.99044C8.84221 7.19721 9.17745 7.20074 9.38853 6.99639L9.38854 6.9964L9.38969 6.99525L9.93749 6.44744L10.4847 6.99461C10.5848 7.09837 10.7223 7.15718 10.8663 7.15852C11.0101 7.15986 11.1491 7.10283 11.2507 7.00076C11.3528 6.8992 11.4099 6.76017 11.4085 6.61626C11.4072 6.47227 11.3484 6.33478 11.2446 6.23463Z"
            fill="#545F7D"
            stroke="#545F7D"
            stroke-width="0.2"
          />
        </svg>
         
        <span
          class="ms-2"
        >
          Blacklist User
        </span>
      </li>
      <li
        class="dropdown-item align-items-center d-flex py-2"
      >
        <svg
          fill="none"
          height="14"
          viewBox="0 0 14 14"
          width="14"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M4.98844 6.61042L4.9883 6.61032C4.16917 5.99661 3.6376 5.01997 3.6376 3.91777C3.6376 2.10402 5.07432 0.625685 6.87096 0.557899L6.87141 0.554956H6.99998C8.85641 0.554956 10.3613 2.06095 10.3613 3.91726C10.3613 5.01894 9.82975 5.99541 9.01128 6.60916C9.01126 6.60917 9.01124 6.60919 9.01122 6.60921L8.8311 6.74442L9.04517 6.81481C9.04518 6.81482 9.04519 6.81482 9.0452 6.81482C9.86977 7.08578 10.6377 7.52127 11.3015 8.10107L11.3015 8.10111C11.4683 8.24689 11.4846 8.49091 11.3393 8.65726L11.339 8.65759C11.1928 8.82395 10.9489 8.83881 10.7831 8.69395L10.8767 8.58681L10.7831 8.69395C9.77092 7.80959 8.48183 7.30916 7.13928 7.27625H7.12746L7.12244 7.27643C7.08096 7.27795 7.03808 7.27953 6.99838 7.27953C6.95947 7.27953 6.9184 7.27802 6.87796 7.27653L6.87038 7.27625H6.85911C4.20631 7.34105 1.91033 9.24286 1.36801 11.8468L1.36796 11.8471C1.32757 12.0393 1.37682 12.2406 1.50067 12.3934L1.5188 12.4157C1.60005 12.5064 1.7639 12.6326 2.00101 12.6326H6.51279C6.73367 12.6326 6.9067 12.8061 6.9067 13.027C6.9067 13.2492 6.733 13.4209 6.51279 13.4209H2.00101C1.56642 13.4209 1.16332 13.2288 0.889327 12.892L0.889302 12.892C0.613528 12.5528 0.507874 12.1163 0.597436 11.6872L4.98844 6.61042ZM4.98844 6.61042L5.16885 6.74525M4.98844 6.61042L5.16885 6.74525M10.6969 12.9018L10.7105 12.9154L10.7147 12.9196L10.8019 13.0068C10.8005 13.008 10.7989 13.0093 10.7971 13.0107C10.7941 13.0131 10.7854 13.0199 10.7729 13.0263C10.7662 13.0295 10.7492 13.036 10.7387 13.0389C10.7222 13.0418 10.6806 13.0409 10.6563 13.0349C10.6421 13.0283 10.624 13.015 10.6075 13.0001C10.6386 12.9701 10.6679 12.938 10.6952 12.904L10.6969 12.9018ZM10.6969 12.9018L10.71 12.9116L10.7445 12.9375L10.7445 12.9375L10.8169 12.9918L10.8159 12.9931L10.8178 12.9909L10.8196 12.9889C10.8199 12.9886 10.8202 12.9882 10.8205 12.9878L10.6993 12.8991C10.6991 12.8994 10.6988 12.8998 10.6985 12.9001L10.6969 12.9018ZM5.16885 6.74525L4.95422 6.81591C4.95422 6.81591 4.95422 6.81591 4.95422 6.81591C4.19092 7.06717 3.47392 7.45838 2.84687 7.97401C1.69729 8.91932 0.899155 10.2389 0.59745 11.6872L5.16885 6.74525ZM4.42376 4.03523V3.91725C4.42376 2.50179 5.5796 1.34523 6.9947 1.34327L6.9947 1.34329H6.99881C8.41557 1.34329 9.57335 2.5001 9.57335 3.91727C9.57335 5.2906 8.48741 6.4205 7.12966 6.48821L7.12964 6.48794L7.12056 6.48894L7.11906 6.48911C7.07768 6.48803 7.03776 6.48748 6.99932 6.48748C6.9736 6.48748 6.94583 6.48798 6.91902 6.48847C6.90572 6.48871 6.89265 6.48895 6.88018 6.48912H6.87966L6.87807 6.48894L6.8781 6.48867L6.86894 6.48821C5.60286 6.42537 4.60496 5.42448 4.47602 4.16989L4.46218 4.03523H4.42376ZM13.057 8.81973L13.049 8.81168L12.9732 8.82387C12.8717 8.84017 12.781 8.89585 12.7176 8.98398L12.7175 8.98412L10.1837 12.5199C10.0909 12.6244 9.97132 12.6515 9.90539 12.6559L9.90482 12.6559C9.83814 12.6606 9.71301 12.6487 9.60694 12.5546L9.60672 12.5543L9.59842 12.5462L7.96615 10.9538C7.80784 10.799 7.56305 10.8035 7.40896 10.9611L7.40895 10.9611C7.25466 11.1189 7.25773 11.3629 7.41486 11.5175L7.41527 11.5179L9.05047 13.1136L9.05046 13.1136L9.05179 13.1148C9.27464 13.3271 9.5732 13.445 9.8793 13.445H9.88291C9.90315 13.445 9.92661 13.445 9.94875 13.4436V13.4437L9.95862 13.443C10.2896 13.4212 10.6004 13.2595 10.8092 13.0015L10.8101 13.0007L10.8149 12.9947L10.82 12.9884L10.8205 12.9878L10.8207 12.9879L10.8223 12.9858L10.8224 12.9858L10.8243 12.9831L13.3589 9.44383C13.359 9.4438 13.359 9.44378 13.359 9.44375C13.4881 9.26358 13.4461 9.02238 13.2683 8.89375C13.2028 8.84626 13.1292 8.82316 13.057 8.81973Z"
            fill="#545F7D"
            stroke="#545F7D"
            stroke-width="0.3"
          />
        </svg>
         
        <span
          class="ms-2"
        >
          Activate User
        </span>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
